source/common/network/io_socket_handle_impl.cc

```diff
@@ -519,16 +519,15 @@ Api::SysCallIntResult IoSocketHandleImpl::shutdown(int how) {
 }
 
 absl::optional<std::chrono::milliseconds> IoSocketHandleImpl::lastRoundTripTime() {
-#ifdef TCP_INFO
-  struct tcp_info ti;
-  socklen_t len = sizeof(ti);
-  if (!SOCKET_FAILURE(
-          Api::OsSysCallsSingleton::get().getsockopt(fd_, IPPROTO_TCP, TCP_INFO, &ti, &len).rc_)) {
-    return std::chrono::milliseconds(ti.tcpi_rtt);
+  Api::EnvoyTcpInfo info;
+  auto result = Api::OsSysCallsSingleton::get().socketTcpInfo(fd_, &info);
+  if (!result.rc_) {
+    ENVOY_LOG_MISC(warn,
```

Comment:
I would make this debug level vs. warn to avoid log spam (or better just drop this entirely and let the caller log/warn, etc. as they desire).

Reply:
sure, fixed in 6de9beb3436f68386bcf19f0d1dd34b1bdffa56b